include/pmacc/eventSystem/events/kernelEvents.hpp

```diff
@@ -185,6 +182,14 @@ namespace exec
                 std::string( " [" ) + m_kernel.m_file + std::string( ":" ) +
                 std::to_string( m_kernel.m_line ) + std::string( " ]" );
 
+            /* alpaka reports errors as exceptions, in the check mode handle it
+             * here while we still have kernel information
+             */
+#if( PMACC_SYNC_KERNEL  == 1 )
+            try
```

Comment:
discussed offline: `try catch` should be moved into the macro `CUDA_CHECK_KERNEL_MSG`

Reply:
Implemented and rebased.